common/workunit/workunit.cpp

```diff
@@ -8142,6 +8195,106 @@ bool CLocalWorkUnit::getDebugValueBool(const char * propname, bool defVal) const
     return p->getPropBool(prop.str(), defVal);
 }
 
+StringBuffer & getProcessLogDateFromLogFileName(const char * processType, const char * logFileNameWithPath, StringBuffer & logDate)
+{
+    //WU xml has a 'log' attribute for each log file used by a process. The code here
+    //tries to parse log date from the 'log' attribute if the attribute contains the
+    //log date information. Take a thor master log as an example (ECLAgent log is
+    //similar). The 'log' attribute looks like '...thormaster.yyyy_mm_dd.log' if the
+    //'log' attribute does contain the log date information. If thor master stores all
+    //of its log lines for a WU into one log file, there is no need for the log file
+    //name to have the 'yyyy_mm_dd' part (10 byte long). The log attribute may look
+    //like '...thormaster.log'.
```

Comment:
>If thor master stores all of its log lines for a WU into one log file there is no need for the log file name to have the 'yyyy_mm_dd' part 

It's just the ZAP file creation that causes a filename without a date stamp in it isn't it?
If we ever changed bare-metal to write to a file per wuid, the file format would change (e.g. to contain the wuid name).